### PR TITLE
fix(DB\smart_scripts):fixed NPC Greymist Coastrunner Fleeing at low h…

### PR DESCRIPTION
**Pull request by Ragelezz**
…p and removed battle stance

## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/1781
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7983

## SOURCE:

- source in the issue

## Tests Performed:

- .go c id 2202 and see that he has a rage bar (which he shouldn't) and won't flee at low hp
- 

## How to Test the Changes:

1. .go c id 2202 and see that he no longer has rage bar and flees at low hp
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1632062753781424000.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632062753781424000');
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 2202);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(2202, 0, 0, 0, 2, 0, 101, 0, 10, 15, 500, 1500, 0, 25, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, '');
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1632062753781424000.sql:5]:
```suggestion
(2202, 0, 0, 0, 2, 0, 101, 0, 10, 15, 500, 1500, 0, 25, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, '');

```
add new line

**Review by Helias — APPROVED** [review]

**Comment by syssneck** [comment]
Something is not right, the fleeing at low hp gets triggered over and over again, and they never come back to me (they flee endlessly):
grafik

**Comment by Ragelezz** [comment]
thanks for feedback, is there any repeat timer (in general) on when a npc should flee again or he just flees once and thats it ?

**Comment by syssneck** [comment]
Take a look at `event_flags`. If you put it at 1, the event can not be repeated.
https://www.azerothcore.org/wiki/smart_scripts#eventflags

**Comment by Ragelezz** [comment]
okay, will do that after work today, thanks

**Comment by jamadaha** [comment]
Tested:
Ran SQL query — Changed one row as expected
Checked that he has no rage bar — He doesn't
Checked for flee at low HP — It doesn't flee, but lies to the face of the player (No fleeing, but writes to chat, indicating that it should)
image

**Comment by Ragelezz** [comment]
Just tested and he runs from player, maybe because you are under water?

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by syssneck** [comment]
It still doesn't work correctly for me. The battle stance is removed, but they do not flee in fear at low hp (it only worked once, most do not flee at all).

> Try on solid ground not water. Also, "run away in fear", so he needs to run away from player (like Fear from a warlock)
For me works :-?

No, this has nothing to do with player abilities. (Most) humanoids flee at low hp.

> I think the issue its not the Flee state but movement type. In water you have move_swim state but on ground you have either move_run or move_walk. 

They are all in water. Fleeing should also work underwater.

**Comment by Ragelezz** [comment]
Saw a post somewhere here about creatures movement underwater, think its related to that, Will look for it and post it here.

**Comment by Nyeriah** [comment]
It seems that this was already tested, labeling accordingly.

File: data/sql/updates/pending_db_world/rev_1632062753781424000.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632062753781424000');
+
+
+
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1632062753781424000.sql:4]:
```suggestion
```

File: data/sql/updates/pending_db_world/rev_1632062753781424000.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 2202);
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1632062753781424000.sql:2]:
```suggestion
DELETE FROM `smart_scripts` WHERE (`entryorguid` = 2202) AND (`source_type` = 0);
```

File: data/sql/updates/pending_db_world/rev_1632062753781424000.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632062753781424000');
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 2202) AND (`source_type` = 0);
+
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(2202, 0, 0, 0, 2, 0, 101, 1, 10, 15, 500, 1500, 0, 25, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, '');
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1632062753781424000.sql:5]:
```suggestion
INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632062753781424000');

DELETE FROM `smart_scripts` WHERE (`entryorguid` = 2202) AND (`source_type` = 0);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(2202, 0, 0, 0, 2, 0, 101, 1, 10, 15, 500, 1500, 0, 25, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, '');
```

File: data/sql/updates/pending_db_world/rev_1632062753781424000.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1632062753781424000');
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 2202) AND (`source_type` = 0);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(2202, 0, 0, 0, 2, 0, 101, 1, 10, 15, 500, 1500, 0, 25, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, '');
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1632062753781424000.sql:5]:
```suggestion
(2202, 0, 0, 0, 2, 0, 101, 1, 10, 15, 500, 1500, 0, 25, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Greymist Coastrunner - On Health Percent - Flee');
```

**Pull request merged**